src/main/java/com/enonic/lib/react4xp/ssr/pool/Renderer.java

```diff
@@ -34,155 +35,84 @@ public class Renderer {
-                    : null;
-
-        } catch (ScriptException s) {
-            throw new RenderException(s);
-        }
-    }
-
-
-    private Map<String, String> runSSR(String entry, String props, List<String> assetsInvolved) {
-
-        String call = "ReactDOMServer.renderToString(" + config.LIBRARY_NAME + "['" + entry + "'].default(" + props  + "));";
+    private Map<String, String> runSSR( String entry, String props, List<String> assetsInvolved )
+    {
+        try
+        {
+            final Invocable invocable = (Invocable) engine;
+            final Object propsJson = invocable.invokeMethod( engine.get( "JSON" ), "parse", props );
+            final Map<String, Object> entryObject =
+                (Map<String, Object>) ( (Map<String, Object>) engine.get( config.LIBRARY_NAME ) ).get( entry );
+            final Function<Object, Object[]> defaultFunction = (Function<Object, Object[]>) entryObject.get( "default" );
 
-        try {
-            String rendered = evalAndGetByKey(engine, call, KEY_HTML);
+            final Object entryWithProps = defaultFunction.apply( new Object[]{propsJson} );
 
-            return Map.of( KEY_HTML, rendered );
+            final String renderedHtml = (String) invocable.invokeMethod( engine.get( "ReactDOMServer" ), "renderToString", entryWithProps );
 
-        } catch (RenderException e) {
+            return Map.of( KEY_HTML, renderedHtml );
+        }
+        catch ( ScriptException e )
+        {
             ErrorHandler errorHandler = new ErrorHandler();
-            String cleanErrorMessage = errorHandler.getCleanErrorMessage(e);
-            LOG.error(
-                    (e.getStacktraceString() == null ? "" : e.getStacktraceString() + "\n") +
-                            errorHandler.getLoggableStackTrace(e, cleanErrorMessage) + "\n\n" +
-                            e.getClass().getSimpleName() + ": " + cleanErrorMessage + "\n" +
-                            "in " + ServerSideRenderer.class.getName() + ".runSSR\n" +
-                            "Entry: '" + entry + "'\n" +
-                            "Assets involved:\n\t" + String.join("\n\t", assetsInvolved) + "\n" +
-                            "Failing call: '" + call + "'\n" +
-                            errorHandler.getSolutionTips());
+            String cleanErrorMessage = errorHandler.getCleanErrorMessage( e );
+            String call = config.LIBRARY_NAME + "['" + entry + "'].default(" + props + ")";
+            LOG.error( ( e.getMessage() + "\n" ) + errorHandler.getLoggableStackTrace( e, cleanErrorMessage ) + "\n\n" +
+                           e.getClass().getSimpleName() + ": " + cleanErrorMessage + "\n" + "in " + ServerSideRenderer.class.getName() +
+                           ".runSSR\n" + "Entry: '" + entry + "'\n" + "Assets involved:\n\t" + String.join( "\n\t", assetsInvolved ) +
+                           "\n" + "Failing call: '" + call + "'\n" + errorHandler.getSolutionTips() );
```

Comment:
While we're at it with avoiding concatenation, this might as well be: 

```
LOG.error(
    "{}\n{}\n\n{}: {}\nin {}.runSSR\nEntry: '{}'\nAssets involved:\n\t{}\nFailing call: '{}'\n{}, 
    e.getMessage(),  
    errorHandler.getLoggableStackTrace( e, cleanErrorMessage ),
    e.getClass().getSimpleName(),
    cleanErrorMessage
    ServerSideRenderer.class.getName(),
    entry,
    String.join( "\n\t", assetsInvolved ),
    call,
    errorHandler.getSolutionTips()
);
```